### ISSUE TITLE
Is LeakSanitizer available on Android

### ISSUE THREAD

**Issue by zhihuawensc** [android-ndk/ndk#431]
--------------------------------------------------------------------------------

#### Description
Not sure if LeakSanitizer is supported on 
on pixel with userdebug build android 7.1.2

#### Environment Details
Platform - OSX 10.12.5
Device - Google Pixel, userdebug build NJH47B android-7.1.2_r17
NDK 15.1.4119039

http://source.android.com/source/building.html
https://medium.com/@raminmahmoodi/build-android-5-0-lollipop-on-osx-10-10-yosemite-441bd00ee77a#.7t5h0uv59

https://docs.google.com/document/d/1Ydq5q9pwJ3bu4Ii2p8KKJ_dZGJNEDUmdUM11va0qHL4/edit#

I was trying to use Address Sanitizer on Anroid, I built the 7.1.2 userdebug version on Pixel.

After run ./asan_device_setup, I was able to catch simple use-after-free, stack-overflow and heap-buffer-overflow. However, ASAN seems cant detect simple memory leaks.

**Comment by DanAlbert** [comment]
I'm actually not sure. LSAN is part of ASAN iirc, but idk if that's true for Android. @eugenis or @stephenhines would know.

**Comment by eugenis** [comment]
Not yet, sorry. See https://github.com/google/sanitizers/issues/379

**Comment by deepankar1994** [comment]
@eugenis  Is there any timeline for Leak Sanitizer support in Android?

**Comment by deepankar1994** [comment]
I'm sure a lot of people are looking forward to this.

**Comment by DanAlbert** [comment]
@stephenhines: is this on the roadmap for the platform/toolchain?

**Comment by stephenhines** [comment]
No, we have no timeline for when LSan will be available in Android.

**Comment by deepankar1994** [comment]
@stephenhines  I have found LSan to be very useful in debugging memory leaks. Is there not enough demand for LSan on Android ?

**Comment by stephenhines** [comment]
It just isn't a priority at this time. We don't have the ability to take on every project that could be helpful.

**Comment by deepankar1994** [comment]
Ok. I understand.

**Comment by axnsan12** [comment]
Judging by https://github.com/google/sanitizers/issues/379#issuecomment-907132213 and https://reviews.llvm.org/D85927 LSAN should now be supported on 30/31+?

Is there any update on the NDK side?

**Comment by stephenhines** [comment]
This is a feature request, and official support requires additional testing if we are going to advertise it in the NDK. I think that would be the biggest remaining blocker from considering this "Fixed".

**Comment by 20083017** [comment]
android still not support lsan yet? hope for a quick answer~

**Comment by DanAlbert** [comment]
The quick answers are on this page. The bug is opened and it's in the "unplanned" milestone.
